lisa/sut_orchestrator/azure/platform_.py

```diff
@@ -1989,13 +1994,21 @@ def _get_deployable_vhd_path(
                     if blob.content_settings:
                         cached_key = blob.content_settings.get("content_md5", None)
                     if original_key == cached_key:
-                        # if it exists, return the link, not to copy again.
-                        log.debug("the sas url is copied already, use it directly.")
-                        return full_vhd_path
+                        # If it exists then check if the blob is available by checking
```

Comment:
It looks the logic just need to adjust, it doesn't need to have a new method. The key thing is even the key is the same, it needs to wait, until the blob is success. If the default timeout 1 hour is too short, we can extend it to 2 hours. But 6 hours to wait is too long.

If the vhd is pending more than 6 hours, it should be deleted, and copy again.

```python
            blobs = container_client.list_blobs(name_starts_with=vhd_path)
            vhd_exists = False
            for blob in blobs:
                if blob:
                    # check if hash key matched with original key.
                    if blob.content_settings:
                        cached_key = blob.content_settings.get("content_md5", None)
                    if _is_stuck_copying():
                        # Delete the stuck vhd.
                        blob_client.delete_blob(delete_snapshots="include")
                        break
                    elif original_key == cached_key:
                        # If it exists, return the link, not to copy again.
                        log.debug("the sas url is copied already, use it directly.")
                        vhd_exists = True
                    else:
                        log.debug("found cached vhd, but the hash key mismatched.")

            if not vhd_exists:
                blob_client = container_client.get_blob_client(vhd_path)
                blob_client.start_copy_from_url(
                    original_vhd_path, metadata=None, incremental_copy=False
                )

            wait_copy_blob(blob_client, vhd_path, log)
```